### DISCUSSION TITLE
Is the uid added by grafana-agent? Do we need it?

### DISCUSSION THREAD

**Discussion by luvpreetsingh** [grafana-cold-storage/agent#5919] [Q&A]
Hi,

I have just noticed that all my metrics have a uid.

```
{
__name__="kube_pod_container_resource_requests",
__replica__="replica-0",
cluster="live-cluster",
container="istio-proxy",
endpoint="http",
instance="x.x.x.x:8080",
job="kube-state-metrics",
namespace="default",
node="live-cluster-lg-workloads-abcdef",
pod="a-deployment-7f686c66-aaa",
project="my-project",
resource="cpu",
service="kube-state-metrics",
uid="ff99bcd0-5a49-4bb7-bb9f-952c5cccc3af",
unit="core"
}

```

Is this uid being added by grafana-agent or the underlying component(kube-state-metrics)?

Since the uid is unique for every metric, I suppose this is putting pressure on my mimir setup due to high cardinality. I am thinking of getting rid of it.

* How do I get rid of it? By relabelings or some setting in grafana-agent?
* Will I be at some dis-advantage if I get rid of it?

Thanks.

**Comment by rfratto** [comment]
A uid label isn't added by default, so it's probably being added as part of your config file being used. Is this a config you got from Grafana Cloud?
